File: src/main/javabak/tile_entities/capabilities/EmptyFluidHandler.java
package redcrafter07.processed.block.tile_entities.capabilities;

import net.neoforged.neoforge.fluids.FluidStack;

public class EmptyFluidHandler implements IFluidHandlerModifiable {
    public static final EmptyFluidHandler INSTANCE = new EmptyFluidHandler();

    @Override
    public void setFluidInTank(int tank, FluidStack stack) {
    }

    @Override
    public FluidStack getFluidInTank(int tank) {
        return FluidStack.EMPTY;
    }

    @Override
    public int getTanks() {
        return 0;
    }

    @Override
    public int getTankCapacity(int tank) {
        return 0;
    }

    @Override
    public boolean isFluidValid(int tank, FluidStack stack) {
        return false;
    }

    @Override
    public int fill(FluidStack resource, FluidAction action) {
        return 0;
    }

    @Override
    public FluidStack drain(FluidStack resource, FluidAction action) {
        return FluidStack.EMPTY;
    }

    @Override
    public FluidStack drain(int maxDrain, FluidAction action) {
        return FluidStack.EMPTY;
    }
}

File: src/main/kotlin/redcrafter07/processed/block/tile_entities/capabilities/EmptyFluidHandler.kt
package redcrafter07.processed.block.tile_entities.capabilities

import net.neoforged.neoforge.fluids.FluidStack
import net.neoforged.neoforge.fluids.capability.IFluidHandler

object EmptyFluidHandler : FluidHandlerModifiable {
    override fun setFluidInTank(tank: Int, stack: FluidStack) = Unit
    override fun getTanks(): Int = 0
    override fun getFluidInTank(p0: Int): FluidStack = FluidStack.EMPTY
    override fun getTankCapacity(p0: Int): Int = 0
    override fun isFluidValid(p0: Int, p1: FluidStack): Boolean = false
    override fun fill(p0: FluidStack, p1: IFluidHandler.FluidAction): Int = 0
    override fun drain(p0: FluidStack, p1: IFluidHandler.FluidAction): FluidStack = FluidStack.EMPTY
    override fun drain(p0: Int, p1: IFluidHandler.FluidAction): FluidStack = FluidStack.EMPTY
}